graylog2-server/src/main/java/org/graylog/scheduler/DBCustomJobDefinitionService.java

```diff
@@ -47,21 +48,24 @@ public DBCustomJobDefinitionService(MongoCollections collections) {
 
     public JobDefinitionDto findOrCreate(JobDefinitionDto dto) {
         var jobDefinitionId = new ObjectId(requireNonNull(dto.id(), "Job definition ID cannot be null"));
+        return executeFindOrCreate(dto, Filters.and(
+                Filters.eq("_id", jobDefinitionId),
+                Filters.eq(f("%s.%s", JobDefinitionDto.FIELD_CONFIG, JobDefinitionConfig.TYPE_FIELD), dto.config().type())
+        ));
+    }
+
+    public JobDefinitionDto findByConfigFieldOrCreate(String configField, String configValue, JobDefinitionDto defaultValue) {
```

Comment:
This method can create multiple job definitions because the filter doesn't include a field with a unique key. See https://www.mongodb.com/docs/manual/reference/method/db.collection.findAndModify/#std-label-upsert-and-unique-index. I think we should stick to the methods we already have in `DBJobDefinitionService` to separately check if a job definition already exists and create one if it is missing. That would make the potential race condition more obvious and doesn't suggest otherwise.